src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,105 @@
+package duke.parser;
+
+import duke.DukeException;
+import duke.task.*;
+import duke.ui.Ui;
+
+public class Parser {
+
+    /**
+     * Parses the user input and executes the corresponding command.
+     *
+     * @param userInput The user input to be parsed.
+     * @param tasks     The TaskList object containing the list of tasks.
+     * @param ui        The Ui object responsible for user interface interactions.
+     * @throws DukeException If an error occurs during parsing or execution of the command.
+     */
+    public static void parse(String userInput, TaskList tasks, Ui ui) throws DukeException {
+        String[] parts = userInput.trim().split(" ", 2);
+        String command = parts[0].toLowerCase();
+
+        switch (command) {
+            case "bye":
+                ui.showGoodbye();
+                System.exit(0);
+                break;
+            case "list":
+                tasks.listTasks();
+                break;
+            case "mark":
+                if (parts.length < 2) throw new DukeException("Mark command needs a task number.");
+                int markIndex = Integer.parseInt(parts[1]) - 1;
+                tasks.markTask(markIndex + 1);
+                break;
+            case "unmark":
+                if (parts.length < 2) throw new DukeException("Unmark command needs a task number.");
+                int unmarkIndex = Integer.parseInt(parts[1]) - 1;
+                tasks.unmarkTask(unmarkIndex + 1);
+                break;
+            case "delete":
+                if (parts.length < 2) throw new DukeException("Delete command needs a task number.");
+                int deleteIndex = Integer.parseInt(parts[1]) - 1;
+                tasks.deleteTask(deleteIndex + 1);
+                break;
+            case "todo":
+            case "deadline":
+            case "event":
+                handleTaskCreation(command, parts, tasks, ui);
+                break;
+            case "find":
+                if (parts.length < 2 || parts[1].isEmpty()) {
+                    throw new DukeException("The keyword for find cannot be empty.");
+                }
+                tasks.findTask(parts[1]);
+                break;
+            default:
+                throw new DukeException("I'm sorry, but I don't know what that means :-(");
```

Comment:
perhaps you can create an Exception that extends DukeException, and call it UnknownInputException so that it is clear